src/Microsoft.DotNet.XHarness.Android/Execution/BaseReportManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.Android.Execution
+{
+    class BaseReportManager
```

Comment:
nit: "Base" usually suggests `abstract` class and I am not sure having `Base` and `Old` as options is intuitive. Maybe `ReportManager` and `OldReportManager` would make more sense? Or maybe we could even include the 23 info in the class name so that we have this knowledge somewhere else than the factory too - `Api23AndOlderReportManager`? It should definitely be conveyed somewhere in the class too (at least a doc header) so that when it's read in isolation, we know what it's for.

Reply:
On a second thought - your classes don't share any logic or common code - so it might be better to rather not inherit them and only create `IReportManager` and force the `DumpBugReport` through that.

If they start sharing something, I would ideally keep implementations as separate leaf children, so we would have
```
         BaseReportManager
	      /            \
OldReportManager      NewReportManager
```
and share the logic.

But I leave this up to you.

Reply:
I added `IReportManager`